pkg/statsd/handler_cloud_test.go

```diff
@@ -261,6 +261,7 @@ type countingHandler struct {
 }
 
 func (ch *countingHandler) DispatchMetric(ctx context.Context, m *gostatsd.Metric) error {
+	m.DoneFunc = nil // Clear DoneFunc because it contains non-predictable variable data
```

Comment:
It is really not obvious why this is needed here

Reply:
It confuses the equality check in the test.  I'll move it to be closer to the use up at line 197 and clarify the comment though.

Reply:
Scratch that, I'll clarify the comment.  Moving it is a bit tricky because it's not pointer based.  Even though it's tests, I don't want to break the "return to pool" semantics.